core/src/main/java/discord4j/core/object/entity/channel/ForumChannel.java

```diff
@@ -0,0 +1,247 @@
+            return value;
+        }
+
+        /**
+         * Translates an integer value to a {@link SortOrder}
+         *
+         * @param value The integer value
+         * @return The SortOrder if known, or the fallback UNKNOWN value
+         */
+        public static SortOrder valueOf(final int value) {
+            /* We could exclude UNKNOWN from the lookup array using
+            Arrays.copyOfRange(SortOrder.values(), 1, SortOrder.values().length) but this would just waste CPU time
+            because if -1 is one day used by Discord we would need anyway to update the code here */
+            for (SortOrder sortOrder : SortOrder.values()) {
+                if (sortOrder.getValue() == value) {
+                    return sortOrder;
+                }
+            }
+            // Fallback to UNKNOWN as default value
+            return SortOrder.UNKNOWN;
+        }
+    }
+
+    /**
+     * Gets the default associated {@link SortOrder} for this forum channel
+     *
+     * @return The {@link SortOrder}, wrapped in an {@link Optional}
+     */
+    public Optional<SortOrder> getSortOrder() {
+        return getData().defaultSortOrder().toOptional()
+            .orElseThrow(IllegalStateException::new) // Mandatory for Forum channels
+            .map(SortOrder::valueOf);
+    }
+
+    @Experimental
+    public enum LayoutType {
+        /**
+         * Internal value only, for unknown values
+         */
+        UNKNOWN(-1),
+        NOT_SET(0),
+        LIST_VIEW(1),
+        GALLERY_VIEW(2);
+
+        /**
+         * Discord represented integer value
+         */
+        private final int value;
+
+        LayoutType(int value) {
+            this.value = value;
+        }
+
+        /**
+         * Gets Discord represented value
+         *
+         * @return Discord integer represented value
+         */
+        public int getValue() {
+            return value;
+        }
+
+        /**
+         * Translates an integer value to a {@link LayoutType}
+         *
+         * @param value The integer value
+         * @return The {@link LayoutType} if known, or the fallback UNKNOWN value
+         */
+        public static LayoutType valueOf(final int value) {
+            // Same as SortOrder, we could exclude the UNKNOWN value
+            for (LayoutType layoutType : LayoutType.values()) {
+                if (layoutType.getValue() == value) {
+                    return layoutType;
+                }
+            }
+            // Fallback to UNKNOWN as default value
+            return LayoutType.UNKNOWN;
+        }
+    }
+
+    /**
+     * Gets the associated {@link LayoutType} for this Forum channel
+     *
+     * @return The associated {@link LayoutType}
+     */
+    public LayoutType getDefaultForumLayout() {
+        return getData().defaultForumLayout().toOptional()
+            .map(LayoutType::valueOf)
+            .orElseThrow(IllegalStateException::new); // Mandatory for Forum channels
```

Comment:
in what cases this can throw a exception if ever return a value?

Reply:
and its not better return the LayoutType.UNKNOWN like in the valueOf where you add this case?

Reply:
I considered the unknown value to any value which is not yet known by Discord4J, not as a fallback when having an inconsistent state regarding our object